internal/heartbeat/heartbeat.go

```diff
@@ -97,6 +104,34 @@ func NewHeartbeatService(dispatcherClient pb.DispatcherClient, configManager *cf
 	}
 }
 
+func (s *Heartbeat) send(data *pb.Data) error {
+	// Just make sure that all send finished
+	s.wg.Wait()
```

Comment:
Why not a mutex?

Reply:
Mutex on external dependencies can be super evil, at the end here, we only need to wait for anything to finish.

Reply:
Actually, when I look at it, there is a deadlock, I think: you call `wg.Wait()` before there was any `wg.Add` so no go routine is unable to proceed to executing the method.

Reply:
OK, scratch that:D If the there was no `Add`, there is nothing to  wait for and it should go through... :facepalm:

Reply:
https://go.dev/play/p/kcwSfkjTfSS

Reply:
In a corner case there might be more go routines sending than 1:
- three go routines: A, B, C
- A increments the counter in wg and starts sending
- B comes and stops at `Wait`
- C comes and stops at `Wait`
- A finishes and calls `wg.Done`
- B and C are released, increment the counter twice and both start sending.

Is that expected? If no, then why not mutex, which will block concurrent execution?